### PR TITLE
Fix unused lambda capture warnings with clang-5

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: packages/Search/test/tstLinearBVH.cpp [hunk 1]
```diff
@@ -303,10 +303,10 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, structured_grid, DeviceType )
...
     double Ly = 100.0;
     double Lz = 100.0;
-    int constexpr nx = 11;
-    int constexpr ny = 11;
-    int constexpr nz = 11;
-    int constexpr n = nx * ny * nz;
+    int nx = 11;
+    int ny = 11;
+    int nz = 11;
+    int n = nx * ny * nz;
```
Comment (aprokop) [inline packages/Search/test/tstLinearBVH.cpp]: Is getting rid of `constexpr` necessary?

Comment (dalg24) [reply packages/Search/test/tstLinearBVH.cpp]: > Necessary

No

We have two options:
* either we keep the const/constexpr declaration and we need an empty lambda capture
* or we get rid of it

Comment (aprokop) [reply packages/Search/test/tstLinearBVH.cpp]: Sorry, I just don't know how things work. Can `constexpr` variables not be captured by lambdas?

Comment (dalg24) [reply packages/Search/test/tstLinearBVH.cpp]: See commit message for 6ce897f

**Pull request merged**